File: frontend/lib/types/course.ts
export interface Course {
  id: string;
  title: string;
  description: string;
  batchIds: string[];
  status: 'active' | 'completed' | 'upcoming';
  // Present on student course listings when the row represents a
  // specific (course, student-batch) pair. A student enrolled in two
  // batches of the same course receives two entries, one per batch.
  batchId?: string;
  batchName?: string;
}

export interface Lecture {
  id: string;
  title: string;
  description: string;
  videoUrl: string;
  duration: string;
  batchId: string;
  batchName: string;
  uploadDate: string;
  order: number;
  courseId: string;
}

export interface CurriculumModule {
  id: string;
  courseId: string;
  title: string;
  description: string;
  order: number;
  topics: string[];
}

export type MaterialFileType = 'pdf' | 'excel' | 'word' | 'pptx' | 'image' | 'archive' | 'other';

export interface CourseMaterial {
  id: string;
  batchId: string;
  batchName: string;
  courseId?: string;
  title: string;
  description?: string;
  fileName: string;
  fileUrl: string;
  fileType: MaterialFileType;
  fileSize: string;
  uploadDate: string;
  uploadedBy: string;
  uploadedByRole: 'course-creator' | 'teacher';
}

File: backend/app/schemas/course.py
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class CourseCreate(BaseModel):
    title: str = Field(max_length=500)
    description: Optional[str] = Field(default=None, max_length=10000)


class CourseUpdate(BaseModel):
    title: Optional[str] = Field(default=None, max_length=500)
    description: Optional[str] = Field(default=None, max_length=10000)
    status: Optional[str] = None


class CourseOut(BaseModel):
    id: uuid.UUID
    title: str
    description: Optional[str] = None
    status: str
    batch_ids: list[uuid.UUID] = []
    # Populated only on student course listings; identifies which specific
    # student-batch this row represents when the student is enrolled in
    # multiple batches of the same course.
    batch_id: Optional[uuid.UUID] = None
    batch_name: Optional[str] = None
    cloned_from_id: Optional[uuid.UUID] = None
    created_by: Optional[uuid.UUID] = None
    created_at: Optional[datetime] = None
    cover_image_url: Optional[str] = None

    model_config = {"from_attributes": True}
